pkg/sql/plan2/build_function.go

```diff
@@ -102,9 +102,16 @@ func getFunctionExprByNameAndAstExprs(name string, exprs []tree.Expr, ctx Compil
 		if err != nil {
 			return nil, err
 		}
+		expr, err = convertValueIntoBool(name, expr)
```

Comment:
you can merge convertValueIntoBool and convertValueIntoBool2 

and move the merged code to getFunctionExprByNameAndPlanExprs()

Reply:
ok